### ISSUE TITLE
ImportError: undefined symbol: __cudaPopCallConfiguration

### ISSUE THREAD

**Issue by rajarshighoshal** [MIC-DKFZ/medicaldetectiontoolkit#48]
I'm new to deep learning stuff and I walk through the readme till the Execute step. Where I'm getting this error. Can anyone please guide me what I'm doing wrong?
![image](https://user-images.githubusercontent.com/22083259/58762289-2da10400-856c-11e9-9451-1f247a1589cd.png)

**Comment by pfjaeger** [comment]
See here #41 . At the moment you have to downgrade to torch=0.4.1 . Sorry for that.

**Comment by rajarshighoshal** [comment]
I am using pytorch version 0.4.1 still getting this error!
![image](https://user-images.githubusercontent.com/22083259/58770592-7fc64180-85cd-11e9-8f8c-f4421b826bd5.png)

**Comment by alexfoo-dw** [comment]
@rajarshighoshal 

I downgraded from CUDA 10.0 to 9.0 and ran these commands from the medicaldetectiontoolkit directory again after downgrading to torch 0.4.1. Fixed my import error.

cd cuda_functions/nms_xD/src/cuda/
nvcc -c -o nms_kernel.cu.o nms_kernel.cu -x cu -Xcompiler -fPIC -arch=[arch]
cd ../../
python build.py
cd ../

cd cuda_functions/roi_align_xD/roi_align/src/cuda/
nvcc -c -o crop_and_resize_kernel.cu.o crop_and_resize_kernel.cu -x cu -Xcompiler -fPIC -arch=[arch]
cd ../../
python build.py
cd ../../

**Comment by rajarshighoshal** [comment]
So it's working for cuda 9? Ok I will try then.

**Comment by sophie-isobel** [comment]
@rajarshighoshal did changing to cuda 9 work for you?  

I am getting a similar Import Error: undefined symbol: __cudaRegisterFatBinaryEnd  

!nvidia-smi returns that I have cuda version 10 and tesla K80 GPU (note i am using google colab).

I have specified arch='sm_37' for the K80 gpu and although I haven't downgraded the cuda version from 10 to 9 as @alexfoodw suggested, I can see that the torch.version.cuda is '9.0.176' for the torch ==0.4.1 I have installed, so the torch should be using cuda 9. 

Would you know why I am having this issue?

**Comment by rajarshighoshal** [comment]
No, I tried after changing to cuda 9. Not working! I'm thinking about
trying with cuda 8! 😅

**Comment by sophie-isobel** [comment]
@rajarshighoshal so I changed to cuda 9 and also changed the default gcc version using the following code, and I am no longer getting this error.

```
!update-alternatives --remove-all gcc 
!update-alternatives --remove-all g++

!update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-5 20
!update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-5 20

!update-alternatives --install /usr/bin/cc cc /usr/bin/gcc 30
!update-alternatives --set cc /usr/bin/gcc

!update-alternatives --install /usr/bin/c++ c++ /usr/bin/g++ 30
!update-alternatives --set c++ /usr/bin/g++

```

**Comment by rajarshighoshal** [comment]
Oh! Okkk!!! Thanks a lot! I will try this!

**Comment by MancaZerovnikMekuc** [comment]
@sophie-isobel  Have you successfully run this on Google Colab? I have CUDA 9.0, pythorch 0.4.1. and changed the gcc as suggested, compiled with 'sm_37' but I am still getting soma Cuda errors.

**Comment by heyitsguay** [comment]
Does this mean medicaldetectiontoolkit will not work with CUDA 10.2 and g++ 7.5.0 with a GTX 1080 (sm_61)? I'm getting the same "undefined symbol: __cudaRegisterFatBinaryEnd" error running python exec.py --mode train on the included toy experiment.

Update: Switched to the torch1x branch and got it working.

**Comment by henbucuoshanghai** [comment]
pytorch is 1.1？  1.4？@h @heyitsguay
